File: content/snippets/cpp/interview_lcci_01_01_is_unique.cpp
#include <array>
#include <iostream>
#include <string>

bool isUnique(const std::string& s) {
    std::array<bool, 256> seen{};
    for (unsigned char ch : s) {
        if (seen[ch]) {
            return false;
        }
        seen[ch] = true;
    }
    return true;
}

int main() {
    std::cout << "leetcode -> " << std::boolalpha << isUnique("leetcode") << '\n';
    std::cout << "abc -> " << std::boolalpha << isUnique("abc") << '\n';
    return 0;
}

File: content/snippets/go/interview_lcci_01_01_is_unique.go
package main

import "fmt"

func isUnique(s string) bool {
	seen := [256]bool{}
	for i := 0; i < len(s); i++ {
		ch := s[i]
		if seen[ch] {
			return false
		}
		seen[ch] = true
	}
	return true
}

func main() {
	fmt.Printf("leetcode -> %t\n", isUnique("leetcode"))
	fmt.Printf("abc -> %t\n", isUnique("abc"))
}
